### ISSUE TITLE
Add delegating serializers for `JsonPatch` and `JsonMergePatch`

### ISSUE THREAD

**Issue by mattnelson** [FasterXML/jackson-datatypes-misc#28]
To simplify the serialization of json-p patch objects, proposing adding delegating serializers for JsonPatch>JsonArray and JsonMergePatch>JsonValue

It takes [extra](https://javaee.github.io/javaee-spec/javadocs/javax/json/JsonPatch.html#toJsonArray--) [steps](https://javaee.github.io/javaee-spec/javadocs/javax/json/JsonMergePatch.html#toJsonValue--) to convert a patch into a type that has a serializer available.

```java
        addSerializer(JsonPatch.class, new StdDelegatingSerializer(new Converter<JsonPatch, JsonArray>() {
            @Override
            public JsonArray convert(JsonPatch value) {
                return value.toJsonArray();
            }
            @Override
            public JavaType getInputType(TypeFactory typeFactory) {
                return typeFactory.constructFromCanonical(JsonPatch.class.getName());
            }
            @Override
            public JavaType getOutputType(TypeFactory typeFactory) {
                return typeFactory.constructFromCanonical(JsonArray.class.getName());
            }
        }));
```

**Comment by cowtowncoder** [comment]
Sounds good to me.
Since this does not require changes to API, could even go in 2.14(.2) patch.

**Comment by cowtowncoder** [comment]
Implemented as suggested; will be included in 2.14.2
